### DISCUSSION TITLE
v0.17.2

### DISCUSSION THREAD

**Discussion by WithoutPants** [stashapp/stash#3048] [Announcements]
# Release Notes
## v0.17.2 - 2022-10-25

Please view the release notes for 0.17.0 if you are upgrading from a version prior to 0.17.0.

### 🐛 Bug fixes
* Fix file rename detection on case-insensitive file systems. ([#3047](https://github.com/stashapp/stash/pull/3047))
* Fix size calculation for symlinks. ([#3046](https://github.com/stashapp/stash/pull/3046))
* Use file base name as title if title is empty in scraper operations. ([#3040](https://github.com/stashapp/stash/pull/3040))
* Fix error when submitting fingerprints from deleted scene. ([#3039](https://github.com/stashapp/stash/pull/3039))
* Fix moved zip file creating duplicate galleries. ([#3036](https://github.com/stashapp/stash/pull/3036))

<hr /><em>This discussion was created from the release <a href='https://github.com/stashapp/stash/releases/tag/v0.17.2'>v0.17.2</a>.</em>

**Comment by philpw99** [comment]
After a scan, most of my scenes have duplicate file entries with different file ids.
Those duplicate entries have all other data the same, like path, size, codec, width, height... etc, but only the file ids and parent folder ids are different.  I don't why they have different parent folder id when their paths are the same.
Is this normal ?

**Reply by WithoutPants** [reply]
Can you provide an example?

**Comment by philpw99** [comment]
I found out a little more about this: 
I have 2 big hard drives, 14TB F: and 8TB G: drive. For some reason, every file located in G: drive has double entries, while all files in F: drive don't. Both hard drive is using exFAT as the file system. G: drive is connected directly to the motherboard with SATA cable, while F: drive is an external USB 3 drive.
For a file located in G: drive, I typed in the playground:
```
{
  findScene(id:407){
    id,
    files{
      id,
      path,
      basename,
      parent_folder_id,
      mod_time,
      size,
      fingerprints{
        type,
        value
      }
      format,
      width,
      height,
      duration,
      video_codec,
      audio_codec,
      frame_rate,
      bit_rate,
      created_at,
      updated_at
    }
  }
}
```
And I will get the double-entry return:
```
{
  "data": {
    "findScene": {
      "id": "407",
      "files": [
        {
          "id": "1545",
          "path": "g:VR\\Western\\Virtual Taboo\\VirtualTaboo.19.02.05.Rebecca.Volpetti.Your.Slutty.Bunny.XXX.VR180.2700p.MP4-VACCiNE[rarbg]\\vac-vt190205rvysb-2700.mp4",
          "basename": "vac-vt190205rvysb-2700.mp4",
          "parent_folder_id": "398",
          "mod_time": "2021-07-08T22:50:16-04:00",
          "size": 2509881139,
          "fingerprints": [
            {
              "type": "oshash",
              "value": "5e84377b78734dd4"
            },
            {
              "type": "phash",
              "value": "dd936dd22d126592"
            }
          ],
          "format": "mp4",
          "width": 5400,
          "height": 2700,
          "duration": 662.36,
          "video_codec": "hevc",
          "audio_codec": "aac",
          "frame_rate": 59.94,
          "bit_rate": 30314526,
          "created_at": "2021-09-10T22:30:33-04:00",
          "updated_at": "2021-09-19T00:36:57-04:00"
        },
        {
          "id": "3147",
          "path": "g:\\VR\\Western\\Virtual Taboo\\VirtualTaboo.19.02.05.Rebecca.Volpetti.Your.Slutty.Bunny.XXX.VR180.2700p.MP4-VACCiNE[rarbg]\\vac-vt190205rvysb-2700.mp4",
          "basename": "vac-vt190205rvysb-2700.mp4",
          "parent_folder_id": "1503",
          "mod_time": "2021-07-08T22:50:16-04:00",
          "size": 2509881139,
          "fingerprints": [
            {
              "type": "oshash",
              "value": "5e84377b78734dd4"
            }
          ],
          "format": "mp4",
          "width": 5400,
          "height": 2700,
          "duration": 662.36,
          "video_codec": "hevc",
          "audio_codec": "aac",
          "frame_rate": 59.94,
          "bit_rate": 30314526,
          "created_at": "2022-11-16T12:04:23-05:00",
          "updated_at": "2022-11-16T12:04:23-05:00"
        }
      ]
    }
  }
}
```
However, if I changed the scene id to 1482, which is located in F: drive, it will return data normally.
```
{
  "data": {
    "findScene": {
      "id": "1482",
      "files": [
        {
          "id": "2599",
          "path": "F:\\Collection\\Western VR\\LustReality\\LustReality.20.05.11.Jia.Lissa.Video.For.My.Horny.Stepdaddy.XXX.VR180.3072p.MP4-Narcos[rarbg]\\narcos-lr-20-05-11-jia-lissa-video-for-my-horny-stepdaddy-vr180-3072p.mp4",
          "basename": "narcos-lr-20-05-11-jia-lissa-video-for-my-horny-stepdaddy-vr180-3072p.mp4",
          "parent_folder_id": "1015",
          "mod_time": "2022-09-17T01:04:56-04:00",
          "size": 2704538897,
          "fingerprints": [
            {
              "type": "oshash",
              "value": "b0ba563a862c53dd"
            },
            {
              "type": "phash",
              "value": "d5a35696e8b85ca1"
            }
          ],
          "format": "mp4",
          "width": 6144,
          "height": 3072,
          "duration": 708.08,
          "video_codec": "h264",
          "audio_codec": "aac",
          "frame_rate": 59.94,
          "bit_rate": 30556524,
          "created_at": "2022-10-12T12:11:30-04:00",
          "updated_at": "2022-10-12T12:11:30-04:00"
        }
      ]
    }
  }
}
```
Before I did the scan after updated to v17.2, I turned off everything in the "Library" settings like this:
<img src="https://user-images.githubusercontent.com/22040708/202253979-a5aa6c17-e0c1-4133-a52c-240772b402ea.png" width=500 />
So the scan will only handle the minimum of data.
I hope the info above will help you find out the reason.

**Reply by WithoutPants** [reply]
In that first query, the paths are subtly different. The first has `g:VR\\...` and the second has `g:\\VR\\...`. If I were to guess, I'd say that you had existing scenes with `g:VR\\...` which is an invalid path, then when you migrated and scanned, it migrated the first, then picked up the latter at the different location and added it to the same scene because it had the same fingerprints.

**Comment by philpw99** [comment]
Yeah, now I see ! 
The `g:VR\\... ` setting is the path inherited by the previous scans by other versions, all the way back to v12.
It never got changed during upgrades for some reason, and I thought that's just the way it is.
